src/hal/buttons.cpp

```diff
@@ -46,12 +62,53 @@ void OswHal::checkButtons(void) {
   } else if (_btn3UpSince < _btn3DownSince) {
     _btn3UpSince = millis();
   }
+
+  //Buttons (Engine)
+  btnIsDown[0] = digitalRead(BTN_1) == LOW;
+  btnIsDown[1] = !digitalRead(BTN_2) == LOW;
+  btnIsDown[2] = !digitalRead(BTN_3) == LOW;
+
+  for(uint8_t i = 0  ; i < 3 ; i++){
+    btnGoneUp[i] = btnLastState[i] == true && btnIsDown[i] == false;
+    btnGoneDown[i] = btnLastState[i] == false && btnIsDown[i] == true;
+    
+    if(btnGoneDown[i]){
+      btnIsDownSince[i] = millis();  
+    }
+    btnLongPress[i] = btnIsDownSince[i] < millis() - DEFAULTLAUNCHER_LONG_PRESS && btnIsDown[i] == true;
```

Comment:
`millis() > btnIsDownSince[i] + DEFAULTLAUNCHER_LONG_PRESS` would protect against overflow issues

Reply:
Yeah we can chance that, when its switched to unisgned :) Will chance that.